### PR TITLE
update optimization doc for BERT related fusions 

### PR DESCRIPTION
**Pull request by tianleiwu**
**Description**: Update the doc to include optimizations for BERT

**Motivation and Context**
- Why is this change required? What problem does it solve?
- If it fixes an open issue, please link to the issue here.

### CODE REVIEW

File: docs/ONNX_Runtime_Graph_Optimizations.md [hunk 1]
```diff
@@ -41,6 +42,15 @@ These optimizations include complex node fusions. They are run after graph parti
 * Matmul Add Fusion
 * Conv Activation Fusion
 * GELU Fusion
+* Layer Normalization Fusion
+
+And BERT related optimizations:
+* Embedding Layer Fusion
+* Attention Fusion
+* Layer Normalization and Skip Connection (with optional Add bias) Fusion
+* GELU Approximiation
```
Comment (KeDengMS) [inline docs/ONNX_Runtime_Graph_Optimizations.md]: >Approximiation [](start = 7, length = 14)

typo

File: docs/ONNX_Runtime_Graph_Optimizations.md [hunk 2]
```diff
@@ -41,6 +42,15 @@ These optimizations include complex node fusions. They are run after graph parti
...
+
+Approximation is used in GELU approximation and Attention fusion. There might be slight difference in result after these optimizations.
```
Comment (askhade) [inline docs/ONNX_Runtime_Graph_Optimizations.md]: Could you elaborate this a little more. For instance add more information on why there could be result diffs and the trade-offs of using this optimization .i.e perf vs accuracy.

Comment (tianleiwu) [reply docs/ONNX_Runtime_Graph_Optimizations.md]: I will add some comments of accuracy impact when result is available later.

File: docs/ONNX_Runtime_Graph_Optimizations.md [hunk 3]
```diff
@@ -32,15 +32,26 @@ These are semantics-preserving graph rewrites which remove redundant nodes and r
...
+| Conv Activation Fusion          | cpu                |                                                             |
+| GELU Fusion                     | cpu or cuda        |                                                             |
+| Layer Normalization Fusion      | cpu or cuda        |                                                             |
+| Embedding Layer Fusion          | cpu or cuda        | Fuse BERT embedding and layer normalization                 |
+| Attention Fusion                | cpu or cuda        | Attention mask has approximation in cuda execution provider |
+| Skip Layer Normalization Fusion | cpu or cuda        | Fuse Add bias, skip connection and layer normalization      |
+| Bias GELU Fusion                | cpu or cuda        | Fuse Add bias and GELU activation                           |
+| GELU Approximation              | cuda               | Erf is approximated by a formula using tanh function        |
+
+For cuda execution provider, approximation is used in GELU approximation and Attention fusion to optimize inference performance of BERT model. There might be slight difference in result.
```
Comment (askhade) [inline docs/ONNX_Runtime_Graph_Optimizations.md]: GELU fusion? (not approximation)

Comment (tianleiwu) [reply docs/ONNX_Runtime_Graph_Optimizations.md]: They are different optimizations. GELU fusion is for cpu and cuda. GELU approximation is applied on the fused node, and it is cuda only.

File: docs/ONNX_Runtime_Graph_Optimizations.md (hunk 2 above)
Comment (tianleiwu) [reply docs/ONNX_Runtime_Graph_Optimizations.md]: I added comment about accuracy impact of approximation. Measurement on SQuAD v1.1 data set shows that the impact can be neglected (F1 score changes from 87.05 to 87.03).

**Pull request merged**